src/main/java/redis/clients/jedis/SslOptions.java

```diff
@@ -0,0 +1,433 @@
+         * @param truststorePassword the truststore password. May be empty to omit password and the truststore integrity check.
+         * @return {@code this}
+         */
+        public Builder truststore(URL truststore, String truststorePassword) {
+
+            Objects.requireNonNull(truststore, "Truststore must not be null");
+
+            return truststore(Resource.from(truststore), getPassword(truststorePassword));
+        }
+
+        /**
+         * Sets the Truststore resource to load trusted certificates. The truststore resource must be supported by
+         * {@link java.security.KeyStore} which is {@link KeyStore#getDefaultType()} by default. The truststore is reloaded on
+         * each connection attempt that allows to replace certificates during runtime.
+         *
+         * @param resource the provider that opens a {@link InputStream} to the keystore file, must not be {@code null}.
+         * @param truststorePassword the truststore password. May be empty to omit password and the truststore integrity check.
+         * @return {@code this}
+         */
+        private Builder truststore(Resource resource, char[] truststorePassword) {
+
+            Objects.requireNonNull(resource, "Truststore InputStreamProvider must not be null");
+
+            this.truststorePassword = getPassword(truststorePassword);
+            this.truststoreResource = resource;
+
+            return this;
+        }
+
+        public Builder insecureTrust(boolean insecureTrust) {
+            this.insecureTrust = insecureTrust;
+            return this;
+        }
+
+        /**
+         * Create a new instance of {@link SslOptions}
+         *
+         * @return new instance of {@link SslOptions}
+         */
+        public SslOptions build() {
+            return new SslOptions(this);
+        }
+
+    }
+
+    /**
+     * A {@link SSLContext} object that is configured with values from this {@link SslOptions} object.
+     *
+     * @return a {@link SSLContext}.
+     * @throws IOException thrown when loading the keystore or the truststore fails.
+     * @throws GeneralSecurityException thrown when loading the keystore or the truststore fails.
+     */
+    public SSLContext createSslContext() throws IOException, GeneralSecurityException {
+
+        final KeyManagerFactory keyManagerFactory;
+        if (keyManagerAlgorithm == null) {
+            keyManagerFactory = null;
+        } else {
+
+            KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+            try (InputStream keystoreStream = keystoreResource.get()) {
+                keyStore.load(keystoreStream, keystorePassword);
+            }
+
+            keyManagerFactory = KeyManagerFactory.getInstance(keyManagerAlgorithm);
+            keyManagerFactory.init(keyStore, keystorePassword);
+
+        }
+
+        final TrustManagerFactory trustManagerFactory;
+        if (insecureTrust) {
+
+            trustManagerFactory = InsecureTrustManagerFactory.INSTANCE;
```

Comment:
why do we need an insecure trust manager at all ?

Reply:
@uglide Yes.

Reply:
@uglide Correct.

Reply:
@tishun We have same features of NONE but the control is separated from SslVerifyMode.

Reply:
Oh, I see, you aimed to separate them in two different configurations - one turns the verification on and off an another (when the config is on) would turn on and off the host verification.